### DISCUSSION TITLE
Advice for running on android smartphone

### DISCUSSION THREAD

**Discussion by anaxonda** [tphakala/birdnet-go#46] [General]
I have a few old android phones lying around. It would be nice to put them to use and not buy another dedicated device. Does anyone have advice running birdnet on an old phone? Should it be possible to run birdnet-go in [termux](https://termux.dev/en/)?

I can imagine a few possibilities:
1)  phone is outside, either connected to external mic or using built in mic
2) phone is inside but connected to wireless mic that is outside 
3) phone is serving audio over network to another device for processing

It would be nice to have a web server running on the phone that could be accessed from other devices. 

I'm aware of a few birdnet android apps, an open source one https://github.com/woheller69/whoBIRD, and one from Cornell: https://play.google.com/store/apps/details?id=de.tu_chemnitz.mi.kahst.birdnet. They seem to be less useful for passive monitoring.

I will try to update this discussion if I learn anything.

**Comment by anaxonda** [comment]
[This discussion](https://github.com/termux/termux-app/issues/1448) may help with difficulties of accessing mic in termux.

**Comment by Shellfishgene** [comment]
3. should be possible with birdnet-go processing a rtsp stream coming from one of the various Android webcam apps?
